crates/bevy_ui/src/texture_slice.rs

```diff
@@ -0,0 +1,177 @@
+// This module is mostly copied and pasted from `bevy_sprite::texture_slice`
+//
+// A more centralized solution should be investigated in the future
+
+use bevy_asset::{AssetEvent, Assets};
+use bevy_ecs::prelude::*;
+use bevy_math::{Rect, Vec2};
+use bevy_render::texture::Image;
+use bevy_sprite::{ImageScaleMode, TextureSlice};
+use bevy_transform::prelude::*;
+use bevy_utils::HashSet;
+
+use crate::{widget::UiImageSize, BackgroundColor, CalculatedClip, ExtractedUiNode, Node, UiImage};
+
+/// Component storing texture slices for image nodes entities with a tiled or sliced  [`ImageScaleMode`]
+///
+/// This component is automatically inserted and updated
+#[derive(Debug, Clone, Component)]
+pub struct ComputedTextureSlices {
+    slices: Vec<TextureSlice>,
+    image_size: Vec2,
+}
+
+impl ComputedTextureSlices {
+    /// Computes [`ExtractedUiNode`] iterator from the sprite slices
+    ///
+    /// # Arguments
+    ///
+    /// * `transform` - the sprite entity global transform
+    /// * `original_entity` - the sprite entity
+    /// * `sprite` - The sprite component
+    /// * `handle` - The sprite texture handle
+    #[must_use]
+    pub(crate) fn extract_ui_nodes<'a>(
+        &'a self,
+        transform: &'a GlobalTransform,
+        node: &'a Node,
+        background_color: &'a BackgroundColor,
+        image: &'a UiImage,
+        clip: Option<&'a CalculatedClip>,
+        camera_entity: Entity,
+    ) -> impl ExactSizeIterator<Item = ExtractedUiNode> + 'a {
+        self.slices.iter().map(move |slice| {
+            let transform =
+                transform.mul_transform(Transform::from_translation(slice.offset.extend(0.0)));
+            let scale = slice.draw_size / slice.texture_rect.size();
+            let mut rect = slice.texture_rect;
+            rect.min *= scale;
+            rect.max *= scale;
+            let atlas_size = Some(self.image_size * scale);
+            ExtractedUiNode {
+                stack_index: node.stack_index,
+                color: background_color.0,
+                transform: transform.compute_matrix(),
+                rect,
+                // TODO: support flip
+                flip_x: false,
+                // TODO: support flip
+                flip_y: true,
```

Comment:
~~I tested to make sure with a modified non symmetrical slice image and we need to not flip the Y to get the expected result.~~

I tested both `flip_y: true` and `flip_y: false` and neither have an effect, but seems like with a modified non symmetrical image the result is flipped on y

Reply:
here is an example of above flip issue.
this is a non symmetrical slice image (note the placement of black spots along the left and bottom of image)
![panel-border-010_flip](https://github.com/bevyengine/bevy/assets/1379590/6e3193bf-4a97-49ab-a055-2bf9854bdd77)
this is the result when this image is used in the ui_texture_slice example (black spot on the bottom is flipped to be on top edge)
![flip_ui_9patch](https://github.com/bevyengine/bevy/assets/1379590/98c7cada-effd-49ea-8ac6-34b3525c3970)

Reply:
Thanks for noticing, I'm fixing it

Reply:
Should be fixed @odecay , it seems UI uses inversed Y coordinates than sprites

Reply:
Your suggestion also allowed me to fix flipping for sprites